ams/api/src/main/java/com/netease/arctic/ams/api/process/OptimizingStage.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.api.process;
+
+/** The stage of the optimizing process. */
+public enum OptimizingStage {
+  FULL_OPTIMIZING("full", true),
+  MAJOR_OPTIMIZING("major", true),
+  MINOR_OPTIMIZING("minor", true),
+  COMMITTING("committing", true),
+  PLANNING("planning", false),
+  PENDING("pending", false),
+  IDLE("idle", false),
+  SUSPENDING("suspending", false),
+  SUBMITTING("submitting", false);
```

Comment:
What is the approximate expected time for a "submitted" state? If certain phases change in seconds or even milliseconds, are we able to monitor, identify and persist these phases in a timely manner?

Reply:
submitting is a stage mainly for external process.  when an  external process is being submitted to resources like yarn or k8s, the coresponding stage is submitting.

Reply:
I have made some changes and comments in Action, please check it